com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/SetBreakpointsRequestHandler.java

```diff
@@ -35,16 +41,31 @@
 import com.microsoft.java.debug.core.protocol.Responses;
 import com.microsoft.java.debug.core.protocol.Types;
 
-public class SetBreakpointsRequestHandler implements IDebugRequestHandler {
+public class SetBreakpointsRequestHandler implements IDebugRequestHandler, IHotCodeReplaceListener {
+
+    private static final Logger logger = Logger.getLogger(Configuration.LOGGER_NAME);
+
     private BreakpointManager manager = new BreakpointManager();
 
+    private IDebugAdapterContext context = null;
+
+    private boolean isInitialized = false;
+
     @Override
     public List<Command> getTargetCommands() {
         return Arrays.asList(Command.SETBREAKPOINTS);
     }
 
     @Override
     public CompletableFuture<Response> handle(Command command, Arguments arguments, Response response, IDebugAdapterContext context) {
+        synchronized (this) {
```

Comment:
Change `getTargetCommands()` to
<pre>
    public List<Command> getTargetCommands() {
           return Arrays.asList(Command.SETBREAKPOINTS, Command.CONFIGURATIONDONE);
    }
</pre>
Then put HCR initialization work in configurationDone handler.
<pre>
   if (command == Command.CONFIGURATIONDONE) {
           ... initialize HCR Provider.
   }
</pre>